packages/guides-restructured-text/src/RestructuredText/Parser/Buffer.php

```diff
@@ -84,4 +84,13 @@ public function clear(): void
     {
         $this->lines = [];
     }
+
+    public function unIndent(int $indentation): void
```

Comment:
I get what you are doing here, but it feels like something that can be forgotten when using this class. Should we add this in the `getLinesString` method and `getLines`? with the default of indenting 1?

Reply:
Mmh, I don't really get what you suggest..

Reply:
I would like to see the effect of this method, be part of the `getLinesString` method and `getLines`

eg. 

```
getLines(int $indentation = 1)
{
   return array_map(function($line) {
            if (strlen($line) >= $indentation) {
                return substr($line, $indentation);
            }
            
            return $line;
      }
   }
}
```
This way we can not forget to remove the indentation, we have multiple places where we use the buffer and strip the first part of the indenting before we start another cycle of parsing the content of a block. Those can be replaced this way.

Reply:
In the very place where I used the unIndent function getLines is not used, just the buffer returned. So we would have to return the information about the depth of the indentation and the buffer only to apply unindent lateron?

Reply:
I have created a list of possible locations to dig into: 

- This method removes the indenting like you do. Based on the first line of a block.
 `\phpDocumentor\Guides\RestructuredText\Parser\Productions\BlockQuoteRule::normalizeLines`
- Here I do something simular for lists
  `packages/guides-restructured-text/src/RestructuredText/Parser/Productions/ListRule.php:107`
- `packages/guides-restructured-text/src/RestructuredText/Parser/Productions/EnumeratedListRule.php:109`
- Definition lists: `packages/guides-restructured-text/src/RestructuredText/Parser/Productions/DefinitionListRule.php:99`

Reply:
Yes, that would work for me. I will turn this into an issue.